src/clusterfuzz/_internal/cron/project_setup.py

```diff
@@ -543,20 +551,32 @@ def create_project_settings(project, info, service_account):
         ccs=ccs).put()
 
 
-def create_pubsub_topics(project):
-  """Create pubsub topics for tasks."""
+def _create_pubsub_topic(name):
+  """Create a pubsub topic and subscription if needed."""
+  client = pubsub.PubSubClient()
+  application_id = utils.get_application_id()
+
+  topic_name = pubsub.topic_name(application_id, name)
+  if client.get_topic(topic_name) is None:
+    client.create_topic(topic_name)
+
+  subscription_name = pubsub.subscription_name(application_id, name)
+  if client.get_subscription(subscription_name) is None:
+    client.create_subscription(subscription_name, topic_name)
+
+
+def create_pubsub_topics_for_untrusted(project):
+  """Create pubsub topics from untrusted sources for tasks."""
   for platform in PUBSUB_PLATFORMS:
     name = untrusted.queue_name(project, platform)
-    client = pubsub.PubSubClient()
-    application_id = utils.get_application_id()
+    _create_pubsub_topic(name)
 
-    topic_name = pubsub.topic_name(application_id, name)
-    if client.get_topic(topic_name) is None:
-      client.create_topic(topic_name)
 
-    subscription_name = pubsub.subscription_name(application_id, name)
-    if client.get_subscription(subscription_name) is None:
-      client.create_subscription(subscription_name, topic_name)
+def create_pubsub_topics_for_queue_id(queue_id, platform):
+  """Create pubsub topics from project configs for tasks."""
+  platform = environment.base_platform(platform)
```

Comment:
No, given the new implementation this is no longer necessary.

However, because of how `untrusted.queue_name()` constructs the queue name, we do need to still pass it both the platform and the queue / device model. 

Updated the logic to account for this.